lib/services/ngsi/ngsiService.js

```diff
@@ -106,6 +107,85 @@ function generateNGSIOperationHandler(operationName, entityName, typeInformation
     };
 }
 
+/**
+ * Generate an operation handler for NGSIv2-based operations (query and update). The handler takes care of identifiying
+ * the errors and calling the appropriate callback with a success or a failure depending on how the operation ended.
+ *
+ * Most of the parameters are passed for debugging purposes mainly.
+ *
+ * @param {String} operationName        Name of the NGSI operation being performed.
+ * @param {String} entityName           Name of the entity that was the target of the operation.
+ * @param {Object} typeInformation      Information about the device the entity represents.
+ * @param {String} token                Security token used to access the entity.
+ * @param {Object} options              Object holding all the information about the HTTP request.
+
+ * @return {Function}                  The generated handler.
+ */
+function generateNGSI2OperationHandler(operationName, entityName, typeInformation, token, options, callback) {
```

Comment:
Two final comment regarding this function (minor):

* Review formatting and indent. For instance, some if clauses lines could be joined (e.g. L136 and L137) and the code would be clearer. Other example: L146-148 are over-idented, compared with L144 (in addition L144 and L145 could be joined).
* Text message use "Received the following request from the CB". However, what we are getting here is the response from CB, not the request. Thus, "Received the following response from the CB" seems to be better.

Reply:
Fixed in fa02910958e0265d1e8417af105f4e940020d08e